src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -976,4 +976,89 @@
       it { expect(response).to redirect_to(project_show_path(apache_maintenance_incident_project)) }
     end
   end
+
+  describe 'POST #move_path' do
+    context 'valid project' do
+      let(:repository) { create(:repository, project: apache_project) }
+      let(:path_element) { create(:path_element, repository: repository) }
+
+      context 'without direction' do
+        it 'raises an exception' do
+          expect { post :move_path, params: { project: apache_project } }.to raise_error ActionController::ParameterMissing
+        end
+      end
+
+      context 'only one path_element' do
+        let(:position) { path_element.position}
+
+        context 'direction up' do
+          before do
+            post :move_path, params: { project: apache_project, repository: repository.id, direction: 'up', path: path_element }
+          end
+
+          it 'goes up' do
+            expect(path_element.reload.position).to eq(position)
+            expect(flash[:notice]).to eq("Path moved up successfully")
+            expect(response).to redirect_to({ action: :index, controller: :repositories, project: apache_project })
+          end
+        end
+
+        context 'direction down' do
+          before do
+            post :move_path, params: { project: apache_project, repository: repository.id, direction: 'down', path: path_element }
+          end
+
+          it 'goes down' do
+            expect(path_element.reload.position).to eq(position)
+            expect(flash[:notice]).to eq("Path moved down successfully")
+            expect(response).to redirect_to({ action: :index, controller: :repositories, project: apache_project })
+          end
+        end
+      end
+
+      context 'three path elements' do
+        let(:path_element_2) { create(:path_element, repository: repository) }
+        let(:path_element_3) { create(:path_element, repository: repository) }
+        let(:path_elements) { [path_element, path_element_2, path_element_3] }
+
+        context 'direction up' do
+          let(:move) {
+            post :move_path, params: { project: apache_project, repository: repository.id, direction: 'up', path: path_elements[1] }
+            path_elements.each { |path_element| path_element.reload }
+          }
+
+          it 'goes up' do
+            move
```

Comment:
Do the move in a before and make separate tests out of the expect like described above